externs/browser/html5.js

```diff
@@ -4286,3 +4288,102 @@ Plugin.prototype.length;
 
 /** @type {string} */
 Plugin.prototype.name;
+
+/** @record */
+function CustomElementInterface() {}
```

Comment:
I really wanted to add these directly to HTMLElement - it significantly simplifies the type information. I just couldn't determine if that was correct from the spec and the Polymer project externs had them separated.

I've switched them to be properties on HTMLElement which completely removes the need for a `CustomElement` type and even a `CustomElementInterface`.